packages/components/src/renderer/index.tsx

```diff
@@ -79,12 +84,15 @@ const defaultRenderer = new Renderer()
 defaultRenderer.register(defaultTypesRenderers)
 
 // TODO: This is for backwards compatibility. Remove in the future
-defaultRenderer.set('custom', ({ module, component, config, payload }) => {
-  if (module === 'extensions' && component === 'Dropdown') {
-    return <Dropdown {...payload} config={config} />
+defaultRenderer.set(
+  'custom',
+  ({ module, component, config, payload }: { module: any; component: any; config: any; payload?: any }) => {
+    if (module === 'extensions' && component === 'Dropdown') {
+      return <Dropdown {...payload} config={config} />
+    }
+    return <Custom module={module} component={component} config={config} />
   }
-  return <Custom module={module} component={component} config={config} />
-})
+)
```

Comment:
I guess this is supposed to be removed as stated by the comment. It would still be nice however to have typings. I had to put any there because of the noImplicitAny setting

Reply:
The payload of a custom component can't be known at build time